src/options.rs

```diff
@@ -0,0 +1,284 @@
+        }
+    }
+}
+
+impl OptionsBuilder {
+    fn setter_enabled(&mut self, x: bool) -> &mut Self {
+        if self.setter_enabled.is_some() {
+            warn!("Setter enabled already defined as {:?}, new value is {:?}.",
+                self.setter_enabled, x);
+        }
+        self.setter_enabled = Some(x);
+        self
+    }
+
+    fn setter_pattern(&mut self, x: SetterPattern) -> &mut Self {
+        if self.setter_pattern.is_some() {
+            warn!("Setter pattern already defined as {:?}, new value is {:?}.",
+                self.setter_pattern, x);
+        }
+        self.setter_pattern = Some(x);
+        self
+    }
+
+    fn setter_public(&mut self, x: bool) -> &mut Self {
+        if self.setter_public.is_some() {
+            warn!("Setter visibility already defined as {:?}, new value is {:?}.",
+                self.setter_public, x);
+        }
+        self.setter_public = Some(x);
+        self
+    }
+
+    fn setter_implicit_options(&mut self, x: bool) -> &mut Self {
+        if self.setter_implicit_options.is_some() {
+            warn!("Setter implicit options already defined as {:?}, new value is {:?}.",
+                self.setter_implicit_options, x);
+        }
+        self.setter_implicit_options = Some(x);
+        self
+    }
+
+    fn getter_enabled(&mut self, x: bool) -> &mut Self {
+        if self.getter_enabled.is_some() {
+            warn!("Getter enabled already defined as {:?}, new value is {:?}.",
+                self.getter_enabled, x);
+        }
+        self.getter_enabled = Some(x);
+        self
+    }
+
+    fn parse_attribute(&mut self, attr: syn::Attribute) {
+        trace!("Parsing attribute {:?}.", attr);
+        if attr.style != syn::AttrStyle::Outer || attr.is_sugared_doc {
+            trace!("Ignoring attribute (outer or sugared doc).");
+            return
+        }
+
+        // e.g. `#[setters]`
+        if let syn::MetaItem::Word(ref ident) = attr.value {
+            match ident.as_ref() {
+                "setters" => {
+                        self.setter_enabled(true);
+                        return
+                    },
+                "getters" => {
+                        self.getter_enabled(true);
+                        return
+                    },
+                _ => {}
+            }
+        }
+
+        // e.g. `#[setters(...)]`
+        if let syn::MetaItem::List(ref ident, ref nested_attrs) = attr.value {
+            match ident.as_ref() {
+                "setters" => {
+                        self.setter_enabled(true);
+                        self.parse_setters_options(nested_attrs);
+                        return
+                    },
+                "getters" => {
+                        self.getter_enabled(true);
+                        self.parse_getters_options(nested_attrs);
+                        return
+                    },
+                _ => {}
+            }
+        }
+        debug!("Ignoring attribute.");
```

Comment:
hm, I see what you mean.

I am open to a discussion. However I think we should try to keep it short, since the purpose of this crate is to avoid boiler plate. One of the next logical steps IMO would be to add options per struct field. I.e. prefix, pattern, visibility, etc. could all be changed on a per-field level. At least it should be possible to ignore individual fields - and if we introduce that, we can also introduce overwriting all the other options.

```rust
#[derive(Builder)]
#[builder_options(setters(pattern = mutable, private = true), getters(public = true)]
struct Foo {
   foo: u32,
   #[builder_options(setter(prefix = with, pattern = immutable, private = false), getter(prefix = has, public = false)]
   bar: u32
}
```

vs. 

```rust
#[derive(Builder)]
#[setters(mutable, private), getters(public)]
struct Foo {
   foo: u32,
   #[setter(prefix = with, immutable, public), getter(prefix = has, private)]
   bar: u32
}
```